### PR TITLE
Update title fields in docs/README.md (release-v0.13.x patch)

### PR DESCRIPTION
**Pull request by eliasnorrby**
:information_source: This PR targets a release branch, not `main`.

# Changes

Patch a previous release branch with the changes introduced in #1095, i.e:

Use more appropriate values for the title fields in the frontmatter of README.md. In the context of the sidebar navigation of the website, for example, 'Overview' is too generic.

See #1095 and tektoncd/website#270 for more background.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [:no_good_man:] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [:no_good_man:] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: cmd/interceptors/main.go [hunk 1]
```diff
@@ -68,7 +68,8 @@ func main() {
 
 	service, err := server.NewWithCoreInterceptors(kubeClient, logger)
 	if err != nil {
-		log.Fatalf("failed to initialize core interceptors: %s", err)
+		log.Printf("failed to initialize core interceptors: %s", err)
+		return
```
Comment (afrittoli) [inline cmd/interceptors/main.go:72]: I don't think this change is meant to be in the PR, I wonder it the cherry-pick picked up too much?

**Review by afrittoli — CHANGES_REQUESTED** [review]
It looks like this is more than titles being changed, could you check if the cherry-pick went ok?

File: cmd/interceptors/main.go (hunk 1 above)
Comment (dibyom) [reply cmd/interceptors/main.go:72]: we need this else the build tests fail due to gocritic issues 😢

Comment (eliasnorrby) [reply cmd/interceptors/main.go:72]: Yeah exactly, we found this out in the v0.12.x patch (#1111). So there's 0fd25a6 to fix the titles, cd80b0b to please gocritic, and 6ddf136 to bump the plumbing version for integration tests to pass.

**Review by dibyom — APPROVED** [review]
/lgtm

**Pull request merged**